homeassistant/components/peblar/entity.py

```diff
@@ -0,0 +1,27 @@
+"""Base entity for the Peblar integration."""
+
+from __future__ import annotations
+
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import PeblarConfigEntry, PeblarMeterDataUpdateCoordinator
+
+
+class PeblarEntity(CoordinatorEntity[PeblarMeterDataUpdateCoordinator], Entity):
+    """Defines a Peblar entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, entry: PeblarConfigEntry) -> None:
+        """Initialize the Peblar entity."""
+        super().__init__(coordinator=entry.runtime_data)
+        self._attr_device_info = DeviceInfo(
+            configuration_url=f"http://{entry.data[CONF_HOST]}",
+            identifiers={(DOMAIN, str(entry.unique_id))},
```

Comment:
the unique id already is a string

Reply:
Nope. It can be `None`

Reply:
Meh, we always know it's a string since in theory every instance has an unique id

Reply:
Yeah I don't like this part either, it just becomes messy quickly other wise.
Good news, this will be replaced later on, as the coordinators will carry the data later (but I didn't want to add multiple coordinators in the first PR).